shoes-core/lib/shoes/download.rb

```diff
@@ -140,9 +142,23 @@ def save_to_file(file_path)
       open(file_path, 'wb') { |fw| fw.print @response.body }
     end
 
-    def download_started(content_length)
-      @content_length = content_length
-      @started        = true
+    def download_started(response)
+      unless @started
+        @started = true
+        @content_length = read_content_length(response)
+
+        @response.status = [response.code, response.message]
+        response.each_header do |key|
+          @response.headers[key] = response[key]
+        end
+
+        mark_progress
```

Comment:
hm, if we wrote a spec directly for it then I'm allfor keeping it. Past us, both in writing and reading probably had some reason for it :)